src/napari_threedee/manipulators/base_manipulator.py

```diff
@@ -1,30 +1,34 @@
 from abc import ABC, abstractmethod
-from typing import Optional, Tuple
+from typing import Optional, Tuple, Type
 
+import napari
 import numpy as np
 from napari.utils.geometry import project_points_onto_plane, rotation_matrix_from_vectors_3d
 from napari.utils.translations import trans
 from napari.viewer import Viewer
 from vispy.scene import Mesh
 from vispy.visuals.transforms import MatrixTransform
 
-from .manipulator_utils import make_translator_meshes, select_rotator, color_lines, make_rotator_meshes
-from ..utils.selection_utils import select_line_segment, select_mesh_from_click
-from ..utils.napari_utils import get_vispy_node, get_napari_visual, add_mouse_callback_safe, remove_mouse_callback_safe
+from .manipulator_utils import make_translator_meshes, color_lines, make_rotator_meshes
+from ..base import ThreeDeeModel
+from ..utils.napari_utils import get_vispy_node, add_mouse_callback_safe, remove_mouse_callback_safe
+from ..utils.selection_utils import select_mesh_from_click
 
 
-class BaseManipulator(ABC):
+class BaseManipulator(ThreeDeeModel, ABC):
     """Base class for manipulators.
 
     To implement:
-        1. Define self._initial_translator_normals in the __init__. This a
-        (Nx3) numpy array containing the normal direction for each of the N
-        translators to be created defined in displayed data coordinates.
-        2. Define self._initial_rotator_normals in the __init__. This a
-        (Nx3) numpy array containing the normal direction for each of the N
-        rotators to be created defined in displayed data coordinates.
-        3. Call the super.__init__() last.
-        4. Implement the drag callback functions
+        - the __init__() should take the viewer as the first argument, the layer
+            as the second argument followed by any keyword arguments.
+            Keyword arguments should have default values.
+        - implement a self._set_initial_translation_vectors() method
+        - implement a self._set_initial_rotator_normals() method
```

Comment:
hope this is self explanatory, let me know if you don't see why this is necessary